### ISSUE TITLE
[Bug]: Openai compatible server with HuggingFaceTB/SmolVLM-Instruct producing smaller outputs than expected

### ISSUE THREAD

**Issue by Sriramkk123** [vllm-project/vllm#10896]
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
PyTorch version: 2.5.1+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 23.10 (x86_64)
GCC version: (Ubuntu 13.2.0-4ubuntu3) 13.2.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.38

Python version: 3.10.15 (main, Oct  3 2024, 07:27:34) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-6.5.0-44-generic-x86_64-with-glibc2.38
Is CUDA available: True
CUDA runtime version: 12.0.140
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 4090
GPU 1: NVIDIA GeForce RTX 4090
GPU 2: NVIDIA GeForce RTX 4090
GPU 3: NVIDIA GeForce RTX 4090

Nvidia driver version: 535.171.04
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      48 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             32
On-line CPU(s) list:                0-31
Vendor ID:                          AuthenticAMD
Model name:                         AMD Ryzen Threadripper PRO 5955WX 16-Cores
CPU family:                         25
Model:                              8
Thread(s) per core:                 2
Core(s) per socket:                 16
Socket(s):                          1
Stepping:                           2
Frequency boost:                    enabled
CPU(s) scaling MHz:                 47%
CPU max MHz:                        7031.2500
CPU min MHz:                        1800.0000
BogoMIPS:                           7985.24
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin brs arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
Virtualization:                     AMD-V
L1d cache:                          512 KiB (16 instances)
L1i cache:                          512 KiB (16 instances)
L2 cache:                           8 MiB (16 instances)
L3 cache:                           64 MiB (2 instances)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-31
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Vulnerable: Safe RET, no microcode
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines; IBPB conditional; IBRS_FW; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-ml-py==12.560.30
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] pyzmq==26.2.0
[pip3] torch==2.5.1
[pip3] torchvision==0.20.1
[pip3] transformers==4.46.3
[pip3] triton==3.1.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-cublas-cu12        12.4.5.8                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.4.127                 pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.4.127                 pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.4.127                 pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.1.0.70                 pypi_0    pypi
[conda] nvidia-cufft-cu12         11.2.1.3                 pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.5.147               pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.6.1.9                 pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.3.1.170               pypi_0    pypi
[conda] nvidia-ml-py              12.560.30                pypi_0    pypi
[conda] nvidia-nccl-cu12          2.21.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.4.127                 pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.4.127                 pypi_0    pypi
[conda] pyzmq                     26.2.0                   pypi_0    pypi
[conda] torch                     2.5.1                    pypi_0    pypi
[conda] torchvision               0.20.1                   pypi_0    pypi
[conda] transformers              4.46.3                   pypi_0    pypi
[conda] triton                    3.1.0                    pypi_0    pypi
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.6.4.post1
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0	GPU1	GPU2	GPU3	CPU Affinity	NUMA Affinity	GPU NUMA ID
GPU0	 X 	SYS	SYS	SYS	0-31	0		N/A
GPU1	SYS	 X 	SYS	SYS	0-31	0		N/A
GPU2	SYS	SYS	 X 	SYS	0-31	0		N/A
GPU3	SYS	SYS	SYS	 X 	0-31	0		N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

LD_LIBRARY_PATH=/home/rendernet/.conda/envs/vllm/lib/python3.10/site-packages/cv2/../../lib64:
CUDA_MODULE_LOADING=LAZY
```

</details>

### 🐛 Describe the bug

The openai compatible server with smovlm producing smaller prompt outputs than directly using smolvlm.

**Smolvlm code:**
```
from transformers import AutoProcessor, AutoModelForVision2Seq
import torch
from PIL import Image
from transformers.image_utils import load_image


DEVICE = "cuda" if torch.cuda.is_available() else "cpu"

processor = AutoProcessor.from_pretrained("HuggingFaceTB/SmolVLM-Instruct")
model = AutoModelForVision2Seq.from_pretrained("HuggingFaceTB/SmolVLM-Instruct",
                                                torch_dtype=torch.bfloat16,
                                                _attn_implementation="flash_attention_2" if DEVICE == "cuda" else "eager").to(DEVICE)



# Load images
image1 = load_image("https://huggingface.co/spaces/HuggingFaceTB/SmolVLM/resolve/main/example_images/rococo.jpg")

detailed_prompt = """You are an expert image captioner. Please provide a detailed description of the image that covers:

Main subjects and their characteristics
Background elements and setting
Colors, lighting, and atmosphere
Notable details or unique features
Spatial relationships between elements
Overall mood or impression
Your description should be at least 100 words long and cover all these aspects comprehensively."""

# Create input messages
messages = [
    {
        "role": "user",
        "content": [
            {"type": "image"},
            {"type": "text", "text": detailed_prompt}
        ]
    },
]

# Prepare inputs
prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
inputs = processor(text=prompt, images=[image1], return_tensors="pt")
inputs = inputs.to(DEVICE)

generated_ids = model.generate(**inputs, max_new_tokens=500)
generated_texts = processor.batch_decode(
    generated_ids,
    skip_special_tokens=True,
)

print(generated_texts[0])
```

**Openai compatible code:**
```
import base64
import argparse
from openai import OpenAI

def encode_image(image_path):
    with open(image_path, "rb") as image_file:
        return base64.b64encode(image_file.read()).decode('utf-8')

def main():
    parser = argparse.ArgumentParser(description='Image captioning using OpenAI API')
    parser.add_argument('--image_url', help='URL of the image to caption')
    parser.add_argument('--api-key', default="abc", help='OpenAI API key')
    parser.add_argument('--api-base', default="http://localhost:8000/v1", help='OpenAI API base URL')
    args = parser.parse_args()

    client = OpenAI(
        api_key=args.api_key,
        base_url=args.api_base,
    )

    detailed_prompt = """You are an expert image captioner. Please provide a detailed description of the image that covers:
1. Main subjects and their characteristics
2. Background elements and setting
3. Colors, lighting, and atmosphere
4. Notable details or unique features
5. Spatial relationships between elements
6. Overall mood or impression

Your description should be at least 100 words long and cover all these aspects comprehensively."""


    try:
        chat_response = client.chat.completions.create(
            model="HuggingFaceTB/SmolVLM-Instruct",
            temperature=0.4,
            top_p=0.8,
            frequency_penalty=1.2,
            messages=[{
                "role": "user",
                "content": [
                    {"type": "text", "text": detailed_prompt },
                    {"type": "image_url", "image_url": {"url": args.image_url}},
                ],
            }],
        )
        print("Caption:", chat_response.choices[0].message.content)
    except Exception as e:
        print(f"Error type: {type(e)}")
        print(f"Error message: {str(e)}")

if __name__ == "__main__":
    main()
```

**Example:**
![image 6](https://github.com/user-attachments/assets/1ac52485-a080-4125-933c-9548df04d864)
**Output produced by smolvlm snippet:**
```
The background of the image is a dense forest, with green trees and plants. The trees are tall and have thick trunks, with branches extending upwards and outwards. The leaves are lush and green, indicating that it is either early spring or late summer. The sunlight is filtering through the trees, creating a dappled pattern on the ground.

The elf is standing in a forest clearing, surrounded by trees. The ground is covered with a layer of leaves and moss, indicating that it is a well-maintained area. The forest is dense, with a variety of plants and trees, suggesting that it is a natural, untouched forest.

The elf is holding a bow and arrow, which are both made of wood. The bow is long and slender, with a curved shape. The arrow is made of wood and has a pointed tip. The elf is aiming at a target, which is not visible in the image.

The elf is standing in a relaxed pose, with her feet shoulder-width apart. Her posture is confident and poised, suggesting that she is ready for action. Her expression is serious and focused, indicating that she is concentrating on her task.

The image has a serene and natural atmosphere, with the forest providing a backdrop that is both beautiful and mysterious. The lighting is soft and diffused, creating a sense of calm and tranquility. The overall impression is one of peace and harmony, with the elf's presence adding a touch of magic and wonder to the scene.

In summary, the image depicts a female elf in a forest setting, dressed in a green dress and holding a bow and arrow. The elf is standing in a forest clearing, surrounded by trees and plants.
```
**Output produced by open-ai compatible server:**
```
An elf woman holding a bow and arrow and an arrow nocked as she stands in a forest.
```

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
The prompt format isn't consistent for the two models. Can you also use `LLM.chat` method for vLLM to eliminate the effect of online serving?

**Comment by Sriramkk123** [comment]
Thanks , making it similar to smolvlm kind worked. 
Regarding the LLM.chat part, we are looking at using openai-compatible servers for all inference as a standard. That's the reason , we are working with different models using vllm and testing out.

**Comment by DarkLight1337** [comment]
I understand. For debugging purposes though, it would be better to use `LLM.chat` so that we can distinguish between problems in vLLM Core and those of the server.

**Comment by DarkLight1337** [comment]
In any case, glad that the problem is solved now!
